infrastructure/src/main/java/org/corfudb/infrastructure/SequencerServer.java

```diff
@@ -145,41 +108,29 @@
      * Handler for this server.
      */
     @Getter
-    private final CorfuMsgHandler handler =
-            CorfuMsgHandler.generateHandler(MethodHandles.lookup(), this);
+    private final CorfuMsgHandler handler = CorfuMsgHandler.generateHandler(MethodHandles.lookup(), this);
 
+    @Getter
+    private final SequencerServerCache cache;
 
     @Getter
     @Setter
     private volatile long sequencerEpoch = Layout.INVALID_EPOCH;
 
-    @Override
-    public boolean isServerReadyToHandleMsg(CorfuMsg msg) {
-        if ((sequencerEpoch != serverContext.getServerEpoch())
-                && (!msg.getMsgType().equals(CorfuMsgType.BOOTSTRAP_SEQUENCER))) {
-            log.warn("Rejecting msg at sequencer : sequencerStateEpoch:{}, serverEpoch:{}, "
-                    + "msg:{}", sequencerEpoch, serverContext.getServerEpoch(), msg);
-            return false;
-        }
-        return true;
-    }
-
     private final ExecutorService executor;
 
-    @Override
-    public ExecutorService getExecutor() {
-        return executor;
-    }
-
     /**
      * Returns a new SequencerServer.
+     *
      * @param serverContext context object providing parameters and objects
      */
     public SequencerServer(ServerContext serverContext) {
         this.serverContext = serverContext;
-        this.opts = serverContext.getServerConfig();
-        this.executor = Executors.newFixedThreadPool(serverContext.getSequencerThreadCount(),
-                new ServerThreadFactory("sequencer-", new ServerThreadFactory.ExceptionHandler()));
+        Map<String, Object> opts = serverContext.getServerConfig();
+        this.executor = Executors.newFixedThreadPool(
+                serverContext.getSequencerThreadCount(),
```

Comment:
The design is based on a single threaded sequencer, having this configuration doesn't make sense really.

Reply:
Right now it's possible to set up more than one thread for a sequencer. Let's discuss if we should delete that parameter. Anyways, that's out of scope for this PR